pop-api/integration-tests/contracts/create_token_in_constructor/lib.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
+
+use ink::prelude::vec::Vec;
+use pop_api::{
+	assets::fungibles::{self as api},
+	primitives::AssetId,
+	StatusCode,
+};
+
+pub type Result<T> = core::result::Result<T, StatusCode>;
+
+#[ink::contract]
+mod create_token_in_constructor {
+	use super::*;
+
+	#[ink(storage)]
+	pub struct Fungible {
+		id: AssetId,
+	}
+
+	impl Fungible {
```

Comment:
Shouldn't this contract also test?
```
 start_destroy
 set_metadata
 clear_metadata
```

Reply:
The sole reason for this contract and its integration test is for the use case where a contract creates an asset on instantiation, and testing that. Not sure whether a separate integration test is required for this because the `create` functionality has a integration test already, however is this in a contract message.

Setting the metadata and the other apis you mention have to be called via contract messages and are tested in the other contract.